MaxText/input_pipeline.py

```diff
@@ -216,7 +216,11 @@ def preprocess_dataset(config: ml_collections.ConfigDict,
       tokenizer.TokenizeOp(sp_tokenizer), num_parallel_calls=AUTOTUNE)
 
   # Set global batch size.
-  batch_size = config.per_device_batch_size * global_mesh.size
+  if config.per_device_batch_size < 1:
+    batch_size = global_mesh.size
```

Comment:
Hmmm this code looks right but it is a little hard to read because the batch size doesn't have the normal meaning. Can you consider how to make this easier for future people to follow?

Reply:
Added a comment to clarify why we use that batch_size calculation.